### DISCUSSION TITLE
Town Population screen (with the well), "Buy Monsters:" window, display all the creatures and let the player select the creatures he want to buy

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#10296] [General]
In fheroes2, it would be nice to let the player select the creatures he want to buy.

In fheroes2:

![Buy monsters fh2 01](https://user-images.githubusercontent.com/43583994/94174009-f9560300-fe94-11ea-96c0-ec91ab3215cc.png)

All available creatures are displayed.
The selection of creatures is memorized for each castle.
If the player ticks the sixth box, the last previously selected box is automatically unchecked.
This screen is for explanatory purposes only.

![Buy monsters fh2 cases 01](https://user-images.githubusercontent.com/43583994/94174157-37532700-fe95-11ea-8864-3ac122542870.png)

**Comment by ihhub** [comment]
Hi @LeHerosInconnu , could you please hold for a week creating issues for UI element locations? Some important bugs like incorrect behavior or missing objects shouldn't be sfopped. We need some time to sort these issues and fix in order to ease the job of developers and for the upcoming events.

Also I noticed you haven't touched evil interface which has a lot of bugs🙂

**Comment by LeHerosInconnu** [comment]
Hello @ihhub,

> Hi @LeHerosInconnu , could you please hold for a week creating issues for UI element locations? Some important bugs like incorrect behavior or missing objects shouldn't be sfopped. We need some time to sort these issues and fix in order to ease the job of developers and for the upcoming events.

I can stop for a week, but there may be an avalanche afterwards. :)

> Also I noticed you haven't touched evil interface which has a lot of bugs

I started by being Good and then I'm going to become Evil. :)

**Comment by LeHerosInconnu** [comment]
Hello @ihhub,
 
> > Hi @LeHerosInconnu , could you please hold for a week creating issues for UI element locations? Some important bugs like incorrect behavior or missing objects shouldn't be sfopped. We need some time to sort these issues and fix in order to ease the job of developers and for the upcoming events.
> 
> I can stop for a week, but there may be an avalanche afterwards. :)

I will post the ones already made last week, there are only five. :)

**Comment by ihhub** [comment]
Hi @LeHerosInconnu , please restart your spamming of issues, especially on Evil interface :)

Only please remember to group needed information into the same issue: for example, position of elements within one window is one issue. Missing animation or element is another issue (as it's logically an independent bug) or changing same messages (even in different windows) is one issue as well. We need this in order to do not make extremely easy issues (like 1 line change) which is time wasting for core team as we need to retest and review before accepting, and to do not make a complicated ones... As a conclusion, it's hard to explain such thing so just use your internal feeling :)

**Comment by LeHerosInconnu** [comment]
@ihhub.

> Hi @LeHerosInconnu , please restart your spamming of issues, especially on Evil interface :)
> 
> Only please remember to group needed information into the same issue: for example, position of elements within one window is one issue. Missing animation or element is another issue (as it's logically an independent bug) or changing same messages (even in different windows) is one issue as well. We need this in order to do not make extremely easy issues (like 1 line change) which is time wasting for core team as we need to retest and review before accepting, and to do not make a complicated ones... As a conclusion, it's hard to explain such thing so just use your internal feeling :)

I doubt that Evil people have much inner feeling. :D
But I will try anyway. :)

**Comment by zenseii** [comment]
> All available creatures are displayed. The selection of creatures is memorized for each castle. If the player ticks the sixth box, the last previously selected box is automatically unchecked. This screen is for explanatory purposes only.
> 
> ![Buy monsters fh2 cases 01](https://user-images.githubusercontent.com/43583994/94174157-37532700-fe95-11ea-8864-3ac122542870.png)

Hi @LeHerosInconnu!

I've been thinking about this and maybe just a small simple button/switch on every creature's dwelling in the well is better than having this specifically in the pop-up dialog?

Or both, since there are two problems you can achieve by having both these options:
1. From the general well screen you would be able to set which creatures you in general want to buy when clicking max.
2. From the dialog you would be able to select which creatures to actually buy of the proposed ones in case you see that you don't have enough money for every available one that you set for max
- This last one could also be made so that if you de-select one of the creatures, the freed up funds could be automatically set to recruit the highest tier ones that the newly freed up funds are enough for to buy.

**Comment by LeHerosInconnu** [comment]
Hello @zenseii,

> I've been thinking about this and maybe just a small simple button/switch on every creature's dwelling in the well is better than having this specifically in the pop-up dialog?
> 
> Or both, since there are two problems you can achieve by having both these options:
> 
> 1. From the general well screen you would be able to set which creatures you in general want to buy when clicking max.
> 2. From the dialog you would be able to select which creatures to actually buy of the proposed ones in case you see that you don't have enough money for every available one that you set for max

Good idea. :)

I think that a system of recruitment priority (independent for each city) should also be put in place.
Example: recruit trolls first, then orcs, then cyclopes, etc.

**Comment by ihhub** [comment]
The purpose of the button is to buy everything at once. If a player wants to buy individual monsters they can always do this one by one in the Well's dialog.
